File: test/matlab_comparison/signal_processing/tf/test_tf.jl
using StructuralVibration, DSP
@usingany GLMakie, MAT

# Load Matlab data for comparison
vars = matread("test/matlab_comparison/signal_processing/tf/test_tf.mat")
Hmat = vec(vars["H"])
fmat = vec(vars["f"])
signal1 = vec(vars["signal1"])
signal2 = vec(vars["signal2"])

fs = 1024
bs = 512
H, f = tfestimation(signal1, signal2, bs, hamming(bs), fs = fs, overlap = 0.)[1:2]

N = length(f)
y = [20log10.(abs.(Hmat[1:N]))'; 20log10.(abs.(H))']

lg = (active = true, entry = ["Matlab", "Julia"])
sv_plot(f, y, xlabel = "Frequency (Hz)", ylabel = "Magnitude (dB)", legend = lg)

File: test/matlab_comparison/signal_processing/tf/test_tf.m
fs = 1024;
t = 0:1/fs:5-1/fs;
signal1 = randn(length(t), 1);
signal2 = cos(2*pi*10*t);
[H, f] = tfestimate(signal1, signal2, 512, 0, [], fs);

plot(f, abs(H))

% save test_tf.mat H f signal1 signal2